tests/inject-variable.pipeline.test.ts

```diff
@@ -109,6 +109,58 @@ describe("pipeline variable injection", () => {
     expect(actual.pipeline).toEqual(expected);
   });
 
+  it("replaces optional variables", async ({ expect }) => {
+    const queryVariables = {
+      address: {
+        description: "",
+        type: "string",
+        path: "$.pipeline[0].$match.wallet",
+        optional: true,
+      },
+      value: {
+        description: "",
+        type: "number",
+        path: "$.pipeline[0].$match.amount",
+        optional: true,
+      },
+      isActive: {
+        description: "",
+        type: "boolean",
+        path: "$.pipeline[0].$match.active",
+        optional: true,
+      },
+    };
+    const pipeline = [
+      {
+        $match: {
+          wallet: "##address",
```

Comment:
For optional variables, would empty strings be better than placeholders like `##address` or `##wallet`? The current format appears meaningful but actually isn't, while empty strings would clearly indicate optionality and encourage developers to check the documentation for valid values.